app/questionnaire/questionnaire_schema.py

```diff
@@ -96,9 +116,17 @@ def get_summary_and_confirmation_blocks(self):
         ]
 
     @staticmethod
-    def get_questions_for_block(block_json):
-        for question_json in block_json.get('questions', []):
-            yield question_json
+    def get_all_questions_for_block(block):
```

Comment:
The naming of this confused me a bit.
From what i can find, this never returns "all" of the questions?
it returns all that are left on the block after the variants have been transformed?

Reply:
or maybe not, i thin it uses it when getting answers on routing path as well as for generating the form

Reply:
No, it will retrieve all of the questions on the block, including each of the variants.